### ISSUE TITLE
Using custom shortcut components

### ISSUE THREAD

**Issue by redbaron76** [GeekyAnts/NativeBase#142]
Hello there. I've built a `NavBar` "shortcut" component that basically builds the `Header` because imho all this structure is great when you want to describe a scene but becomes quickly verbose when you have to repeat it over and over in all the scenes of the project. And in my case, using Redux as state container, would be faster to manage the `Header` in a more programmatic way.

The component's signature is as follows:

```
NavBar.propTypes = {
    title:React.PropTypes.string,
    left:React.PropTypes.shape({
        icon: React.PropTypes.string,
        label: React.PropTypes.string,
        onPress:React.PropTypes.func,
    }),
    right:React.PropTypes.shape({
        icon: React.PropTypes.string,
        label: React.PropTypes.string,
        onPress:React.PropTypes.func,
    })
};
```

and it renders the `Header` component with three custom methods for left/right buttons and the title

```
render() {
        return (
            <Header>
                {this._renderLeftButton()}
                {this._renderTitle()}
                {this._renderRightButton()}
            </Header>
        );
    }
```

but, of course, looking at the code, the `Container` component expects an explicit `Header` component as one of its children and it doesn't render it. (Isn't possibile to avoid this behavior? If it's the Header, render it as expected otherwise just render the "unexpected" component... - up to developer what is going on there)

I noticed also that the `Container` component is just a `View` with `flex:1`... so I replaced the `Container` with a simple `<View style={{flex:1}}>` and now my custom `NavBar` component works **BUT** I have to return always something from `this._renderLeftButton()` and `this._renderRightButton()` (can't return null i.e.) otherwise I got an `Cannot read property 'props' of undefined`... so I basically can't decide when and where showing my buttons.

Any suggestion about this? Am I missing something?

This is a great project, well documented and a big help for developers but probably, in my opinion, it lacks a little bit in terms of flexibility, especially in projects one step above.

**Comment by dunnock** [comment]
Same, I would like to customize Icons in the code like below:

```
const errorStyle = {color: "#bb0033"}
const validStyle = {color: "#33bb00"}
const ValidationIcon = (props) => <Icon
                  key={props.key}
                  name={ props.text === "" ? 'ios-alert' : 'ios-checkmark-circle'}
                  style={ props.text === "" ? errorStyle : validStyle }
                />

```

**Comment by flyandi** [comment]
+1 on this. It doesn't seem to work without rewriting the Container element.

**Comment by sanketsahusoft** [comment]
Fixed with NativeBase 2.0
